example/src/main/java/co/infinum/example/MainActivity.kt

```diff
@@ -6,106 +6,163 @@ import android.content.pm.PackageManager
+    private var isRecording = false
+    private var lensFacing = CameraSelector.LENS_FACING_BACK
+//    private var settingsAdapter = SettingsAdapter(listOf())
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_main)
-        AppCompatDelegate.setCompatVectorFromResourcesEnabled(true)
-        initGoldenEye()
+        setContentView(R.layout.main_activity2)
+        if (allPermissionsGranted()) {
+            startCamera()
+        } else {
+            ActivityCompat.requestPermissions(
+                this, REQUIRED_PERMISSIONS, REQUEST_CODE_PERMISSIONS
+            )
+        }
+
+        outputDirectory = getOutputDirectory()
+        cameraExecutor = Executors.newSingleThreadExecutor()
         videoFile = File.createTempFile("vid", "")
+
         initListeners()
+        setUpPinchToZoom()
     }
 
     private fun initListeners() {
-        settingsView.setOnClickListener {
-            prepareItems()
-            settingsRecyclerView.apply {
-                visibility = View.VISIBLE
-                layoutManager = LinearLayoutManager(this@MainActivity)
-                adapter = settingsAdapter
-            }
-        }
+//        settingsView.setOnClickListener {
+//            prepareItems()
+//            settingsRecyclerView.apply {
+//                visibility = View.VISIBLE
+//                layoutManager = LinearLayoutManager(this@MainActivity2)
+//                adapter = settingsAdapter
+//            }
+//        }
 
         takePictureView.setOnClickListener { _ ->
-            goldenEye.takePicture(
+            takePicture(
                 onPictureTaken = { bitmap ->
                     if (bitmap.width <= 4096 && bitmap.height <= 4096) {
                         displayPicture(bitmap)
                     } else {
                         reducePictureSize(bitmap)
                     }
-                },
-                onError = { it.printStackTrace() }
+                }
             )
         }
 
         recordVideoView.setOnClickListener { _ ->
             if (isRecording) {
                 isRecording = false
                 recordVideoView.setImageResource(R.drawable.ic_record_video)
-                goldenEye.stopRecording()
+                videoCapture?.stopRecording()
             } else {
                 startRecording()
             }
         }
 
         switchCameraView.setOnClickListener { _ ->
-            val currentIndex = goldenEye.availableCameras.indexOfFirst { goldenEye.config?.id == it.id }
-            val nextIndex = (currentIndex + 1) % goldenEye.availableCameras.size
-            openCamera(goldenEye.availableCameras[nextIndex])
+            lensFacing =
+                if (lensFacing == CameraSelector.LENS_FACING_BACK) CameraSelector.LENS_FACING_FRONT else CameraSelector.LENS_FACING_BACK
+            CameraX.getCameraWithLensFacing(lensFacing)
+            recreateCamera()
```

Comment:
1. I don't think so, because you can only get front or back camera with camera selector https://stackoverflow.com/questions/61748470/android-camerax-switch-between-multiple-front-cameras
2. I don't think so, it looks like that GoldenEye just has a list of available cameras (I tried it debugging and it is also two) and goes over them. So I think camerax has advantage in this case because you know which camera is which, and with goldenEye you are speculating I guess

Reply:
You can, `goldenEye.availableCameras.find { it.facing == Facing.BACK }`.

I suppose, in this case, CameraX is simpler, but more limited.